### DISCUSSION TITLE
Creating wrapper on top of fabricjs. Need your feed back.

### DISCUSSION THREAD

**Discussion by AmitDigga** [fabricjs/fabric.js#8681] [Show and tell]
## Library (In development, will be refactored heavily)
[Fabric Extended Editor](https://github.com/AmitDigga/ReactFabricEditor).

https://user-images.githubusercontent.com/7884106/217792892-97df28a5-3d56-46ed-bfb5-4594c6832901.mp4

I need this functionality for personal use. It should be useful for others. 

### Things I have solved
1. After working with fabricjs, everything is savable.
2. There is a option to undo.
3. There is a parent child relationship. Where moving parent should affect children and grandchildren too. (other transformation not addressed as of now)
4. There is a option to store data on every object (kind of meta data). Creating and Updating metadata is also savable and undoable.
5. This is a `plugin` based system, so new plugins can be added. We can activate plugins like a click on the button. These will disable every other plugin and activate the clicked one. From there it can listen to all fabric events and make appropriate actions.
6. Similar to plugin, there are `actions` too.
7. There is class `Property` which is there for two way data binding. (for  `fabric.object` data or its meta data)

### Goals in mind
1. Make this library separate from react.
2. Improve plugin and property systems.
3. Add missing functionality

### Need feedback on
1. Is it good enough to be developed upon?
2. Will you use it?
3. Things which i am missing (like other library or stuffs).
4. How to make it more friendly and usable. 
5. Proper use of rxjs.

**Comment by ShaMan123** [comment]
I like the command interface, I want fabric to have a centralized canvas state.
Code looks good.
The plugin interface is similar to #8476 - we are going to redo the brush API and transform it to something like a plugin. 
You are welcome to contribute to the repo

#7670 
#8299

**Reply by AmitDigga** [reply]
I went through these, its not easy to grasp things. Any easy way to get the basics? Any pull requests and commits to know how things work in fabric. So I can start contributing.

**Reply by ShaMan123** [reply]
Read [CONTRIBUTING](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md) and tell me if/what needs to be added there.
What to you wish to PR? Then I can point out an entry point.
Basically you follow https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#-fixing-bugs
change something, run the test suite and add more tests if needed and PR.
You can start with issues labelled as [beginner-friendly](https://github.com/fabricjs/fabric.js/issues?q=is%3Aopen+is%3Aissue+label%3Abeginner-friendly)

**Reply by ShaMan123** [reply]
looking at [beginner-friendly](https://github.com/fabricjs/fabric.js/issues?q=is%3Aopen+is%3Aissue+label%3Abeginner-friendly) there isn't too much juice there 
#7979 was a good issue that turned into #8134 and was abandoned. Needs finishing up.

**Reply by ShaMan123** [reply]
https://github.com/fabricjs/fabric.js/issues/8679 maybe

**Reply by ShaMan123** [reply]
https://github.com/fabricjs/fabric.js/issues/8608
https://github.com/fabricjs/fabric.js/issues/8588

**Reply by ShaMan123** [reply]
https://github.com/fabricjs/fabric.js/issues/8465

**Reply by ShaMan123** [reply]
https://github.com/fabricjs/fabric.js/issues/8042
